### PR TITLE
Eligibility - Claim on behalf

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3002

### Change description
Added a page asking if the claim is being made on the behalf of the claimant e.g. by a lawyer

### Developer self-QA run statement

- [ X] I have clicked through the running application to see if all changes I made actually work.

### Corresponding Integration Test Changes ###
https://github.com/hmcts/cmc-integration-tests/pull/88

### Does this PR introduce a breaking change?

- [ ] Yes
- [ X] No

### CODE REVIEW

File: src/main/features/claim/form/models/eligibility/eligibility.ts [hunk 1]
```diff
@@ -33,20 +36,23 @@ export class Eligibility {
...
-    claimValue?: ClaimValue,
-    singleClaimant?: YesNoOption,
-    singleDefendant?: YesNoOption,
-    eighteenOrOver?: YesNoOption,
-    governmentDepartment?: YesNoOption,
-    helpWithFees?: YesNoOption,
-    claimIsForTenancyDeposit?: YesNoOption
-  ) {
+  constructor (claimOnBehalf?: YesNoOption,
+               claimantAddress?: YesNoOption,
```
Comment (timja-kainos) [inline src/main/features/claim/form/models/eligibility/eligibility.ts]: undo this formatting change please, the formatting was better before

File: src/main/features/claim/form/models/eligibility/eligibility.ts [hunk 2]
```diff
@@ -33,20 +36,23 @@ export class Eligibility {
   @IsIn(YesNoOption.all(), { message: ValidationErrors.YES_NO_REQUIRED, groups: [ValidationGroups.HELP_WITH_FEES] })
   helpWithFees?: YesNoOption
 
-  @IsIn(YesNoOption.all(), { message: ValidationErrors.YES_NO_REQUIRED, groups: [ValidationGroups.CLAIM_IS_FOR_TENANCY_DEPOSIT] })
```
Comment (radoslawn-kainos) [inline src/main/features/claim/form/models/eligibility/eligibility.ts]: It might be a good idea to keep unrelated changes, especially ones that don't bring considerable improvements, out of the PR. In the long run, it would make it much easier to track down *why* something in the code has been changed. It's not a dealbreaker, but at least consider the reasons.

File: src/main/features/claim/helpers/eligibility/validationGroups.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export class ValidationGroups {
+  static readonly CITIZEN_CLAIMANT: string = 'citizen_claimant'
```
Comment (radoslawn-kainos) [inline src/main/features/claim/helpers/eligibility/validationGroups.ts]: Without any context, what does a name like `CITIZEN_CLAIMANT` tell you? My first thought would be that it means that the claimant is a citizen. Also, please use consistent naming - in other places it's called `claimOnBehalf`, so I'd just use that.

File: src/main/features/claim/views/eligibility/claim-on-behalf.njk [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{% from "form.njk" import csrfProtection, errorSummary, radioGroup, saveAndContinueButton %}
+
+{% set title = t('Citizen Claimant | Money claim') %}
+
+{% block content %}
+
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <h1 class="heading-large">{{ t('Are you making this claim, as a legal professional on behalf of someone else?') }}</h1>
+      <p>{{ t('You cant use this service if this claim is not for your own use') }}</p>
```
Comment (radoslawn-kainos) [inline src/main/features/claim/views/eligibility/claim-on-behalf.njk]: Typo: "cant" -> "can’t"

File: src/main/features/claim/helpers/eligibility/validationGroups.ts (hunk 3 above)
Comment (jacmurphy) [reply src/main/features/claim/helpers/eligibility/validationGroups.ts]: Citizen claimant is valid classification for an eligible claim.
Claim on behalf is not a valid classification, it is a discriminator for eligibility. 
I agree with your point about consistency but CLAIM_ON_BEHALF does not fit in with how the ValidationGroup should be named.

ValidationGroup.ts is the opposite of notEligibleReasons.ts. 
So using a different example:
Valid = OVER_18 - "I am OVER_18. Yes - you can use the service" - *Valid reason is true*
Invalid = UNDER_18 - "I am UNDER_18. No - I  can not use the service" - *Invalid reason is true*

If we applied the same logic, we would have
Valid = CLAIM_ON_BEHALF- "This is a CLAIM_ON_BEHALF. No - but you cant use the service" - *valid reason is FALSE*
Valid = CITIZEN_CLAIMANT- "This is a CITIZEN_CLAIMANT. Yes - as nobody is claiming on your behalf, you are eligible" *valid reason is TRUE*

Invalid = CLAIM_ON_BEHALF - "This is a CLAIM_ON_BEHALF. Yes. You can not use the service - *invalid reason is true*

File: src/main/features/claim/form/models/eligibility/eligibility.ts (hunk 2 above)
Comment (jacmurphy) [reply src/main/features/claim/form/models/eligibility/eligibility.ts]: That was done automatically by intellJ. The previous dev just didn't auto format the lines so when i did i changed their allignment

File: src/main/features/claim/form/models/eligibility/eligibility.ts (hunk 1 above)
Comment (jacmurphy) [reply src/main/features/claim/form/models/eligibility/eligibility.ts]: As discussed: this is intelliJ's auto formatting

File: src/main/features/claim/helpers/eligibility/validationGroups.ts (hunk 3 above)
Comment (jacmurphy) [reply src/main/features/claim/helpers/eligibility/validationGroups.ts]: Talked to Damian about it and its been changed now. I took validationGroups to mean reasons why it was valid. Just a misunderstanding on my part. Thanks for catching this

File: src/main/features/claim/form/models/eligibility/eligibility.ts (hunk 1 above)
Comment (radoslawn-kainos) [reply src/main/features/claim/form/models/eligibility/eligibility.ts]: @jacmurphy How is this relevant? IntelliJ auto-formatting rules are not gospel. If your IDE is making random changes throughout the code, then configure it not to.

Comment (jacmurphy) [reply src/main/features/claim/form/models/eligibility/eligibility.ts]: @radoslawn-kainos As discussed with @timja-kainos, it was auto formatted and he didn't mind which one was picked in the end.

**Comment by jacmurphy** [comment]
Created a new PR as a result of a git issue with a detached head
https://github.com/hmcts/cmc-citizen-frontend/pull/348
